Move Android E2E SauceLabs tests from CircleCI to Buildkite

mokagio opened the pull request:
Final automation in the CircleCI to Buildkite migration. Builds on top of #6320 but only to keep the edits to the pipeline configurations sequential; the changes are similar but independent.

Notice CI is green for the canary, however, there's a failure in the full suite:

<img width="1193" alt="image" src="https://github.com/wordpress-mobile/gutenberg-mobile/assets/1218433/4bd0f2c0-898f-4744-b2b0-33db87d1b6f6">

The full suite failures seem to be "legit". The tests run in Sauce Labs and most of them passed, the setup seems solid.

![image](https://github.com/wordpress-mobile/gutenberg-mobile/assets/1218433/a2e66934-56d7-4a48-8f1a-f69b9627ef7d)



## Next steps

- Audit Slack reporting to make sure nothing got lost

## PR submission checklist

- [x] I have considered adding unit tests where possible. – N.A.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary. – N.A.

mokagio committed:
Add keys to Android native packages unit tests steps in Buildkite

mokagio committed:
Move automation to run Android E2E tests from CircleCI to Buildkite

mokagio committed:
Add missing bundle creation step for Android E2E and always bundle first

dcalhoun commented on .buildkite/commands/test-android.sh:
```diff
@@ -0,0 +1,69 @@
+    INPUT="${1-}"
+done
+
+echo '--- :node: Setup Node depenendencies'
```
Merely grammatical suggestion, feel free to disregard. This could be applied to `test-ios.sh` too, if desired. 
```suggestion
echo '--- :node: Set up Node depenendencies'
```

mokagio replied:
I appreciated this grammatical suggestion. Keep them coming!

dcalhoun commented on .buildkite/commands/build-android.sh:
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash -eu
+
+echo "--- :npm: Install Node dependencies"
```
If is "install" is intentional, that is completely fine and please disregard this. I just noted that we currently mix the usage of "install" and "set up." 
```suggestion
echo "--- :npm: Set up Node dependencies"
```

dcalhoun commented on .buildkite/commands/test-android.sh:
```diff
@@ -0,0 +1,69 @@
+      exit $TESTS_EXIT_CODE
+    fi
+
+    echo "For more details about the failed tests, check the Buildkite annotation, the logs under the '$SECTION' section and the tests results in the artifacts tab."
```
I really appreciate the thought you put into helping others debug unexpected failures. 💯

dcalhoun reviewed (approved):
Thank you so much! 

> The full suite failures seem to be "legit". The tests run in Sauce Labs and most of them passed, the setup seems solid.

I triggered a retry for the CI task. the `socket hang up` is an error that occurs at times. Not that we should ignore it, but I do not believe it represents a regression in the product itself. 

> Audit Slack reporting to make sure nothing got lost

This would be great. I do feel we have gaps here.

mokagio committed:
Use "Set up" over "Setup" in some CI step names

Co-authored-by: David Calhoun <github@davidcalhoun.me>

mokagio committed:
Use "Set up" over "Setup" in remaining CI step names

mokagio committed:
Use `android` queue instead of `android-staging` for E2E tests

Now that everything has been validated as working correctly on staging
and that the production queue has been upgraded, we can move to it.

mokagio commented:
Closed in favor of #6521
